src/Dumplie/Application/Metadata/MetadataAccessObject.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Metadata;
+
+use Dumplie\Application\Exception\Metadata\HydrationException;
+use Dumplie\Application\Exception\Metadata\InvalidArgumentException;
+use Dumplie\Application\Exception\Metadata\NotFoundException;
+use Dumplie\Application\Metadata\Schema\TypeSchema;
+use Dumplie\Domain\SharedKernel\Exception\InvalidUUIDFormatException;
+
+final class MetadataAccessObject
+{
+    /**
+     * @var Storage
+     */
+    private $storage;
+
+    /**
+     * @var TypeSchema
+     */
+    private $typeSchema;
+
+    /**
+     * @var Hydrator
+     */
+    private $hydrator;
+
+    /**
+     * @param Storage $storage
+     * @param TypeSchema $typeSchema
+     * @param Hydrator $hydrator
+     */
+    public function __construct(Storage $storage, TypeSchema $typeSchema, Hydrator $hydrator)
+    {
+        $this->storage = $storage;
+        $this->typeSchema = $typeSchema;
+        $this->hydrator = $hydrator;
+    }
+
+    /**
+     * @param array $criteria
+     * @return Metadata|null
+     * @throws \Dumplie\Application\Exception\Metadata\HydrationException
+     */
+    public function findBy(array $criteria = [])
+    {
+        $data = $this->storage->findBy($this->typeSchema->name(), $criteria);
+
+        if (count($data) === 0 || !array_key_exists('id', $data)) {
+            return ;
```

Comment:
If you are talking about space between ``return`` and ``;`` then I think it's perfectly fine.